### DISCUSSION TITLE
Custom histogram aggregation possible?

### DISCUSSION THREAD

**Discussion by MichaelLogutov** [open-telemetry/opentelemetry-dotnet#3156] [Q&A] [answered]
Hello.
Is it possible to write custom histogram aggregation like, for example, [victoria metrics histogram](https://valyala.medium.com/improving-histogram-usability-for-prometheus-and-grafana-bc7e5df0e350)? Or the only way right now is to emulate this with a dynamic growing list of counter instruments?

**Comment by cijothomas** [comment, marked as answer]
As of today, there is no ability to write custom aggregators. There is no spec written for it yet, so naturally if there is a spec requirement, then only it'll be added.

**Reply by MichaelLogutov** [reply]
Thanks.
I saw aggregation enum and thought that using enum is not flexible in terms of extendability for the future. So maybe you should look into that while you're at RC?

**Reply by cijothomas** [reply]
Could you point to the specific class/enum? Aggregation is not exposed anywhere to the user (internal only), so if there is a "leak", would like to know.

**Reply by MichaelLogutov** [reply]
Ah, yeah - my bad. Missed that MetricStreamConfiguration.Aggregation is internal. Ok then.
